src/geometry/ArborX_Box.hpp

```diff
@@ -64,6 +66,15 @@ struct Box
   Point<DIM, Coordinate> _max_corner;
 };
 
+template <typename T, std::size_t N>
+#if KOKKOS_VERSION >= 40400
+KOKKOS_DEDUCTION_GUIDE
+#else
+KOKKOS_FUNCTION
+#endif
+Box(T const (&)[N], T const (&)[N])
+    -> Box<N, std::conditional_t<std::is_integral_v<T>, double, T>>;
```

Comment:
```suggestion
    -> Box<N, std::enable_if_t<std::is_integral_v<T>, double>>;
```
Not tested but it feels like we would potentially enable deductions we did not mean to with your version

Reply:
I think you version would not deduce `Box{{float}, {float}}`.

Reply:
I thought we were only enabling CTAD for arithmetic types.  Is that not what the intent was?

Reply:
I had a dual intent: enable conversion for arithmetic types, and allow CTAD for a single type.

Reply:
I think I understand your concern. However, I would not want `Point(0, 0, 0)` to produce `Point<3, int>`.